### DISCUSSION TITLE
@formily/reactive-react 如何配合shallowEqual做重复渲染优化呢？

### DISCUSSION THREAD

**Discussion by luanwulin** [alibaba/formily#3564] [Q&A 2.x] [answered]
@formily/reactive-react结合observer可以做到组件的更新，但是其他的地方可能也会改动到这个model导致不应该的渲染。

observer如何配合memo拿到某个字段上一次的数据，做重复渲染优化呢？

**Comment by janryWang** [comment, marked as answer]
为啥要整这玩意，reactive本身就是基于你依赖的数据变化才会重新渲染，怎么可能会出现重复渲染的问题
